p2p/pkg/crypto/ecdh.go

```diff
@@ -0,0 +1,111 @@
+package crypto
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+)
+
+// GenerateKeyPairBN254 returns a BN254 private key (fr.Element) and the
+// corresponding public key in G1 affine form.
+func GenerateKeyPairBN254() (sk *fr.Element, pk *bn254.G1Affine, err error) {
+	// 1) Generate random secret in Fr
+	sk = &fr.Element{}
+	_, err = sk.SetRandom()
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate random secret: %w", err)
+	}
+
+	// 2) Retrieve the G1 generator (1,2) from the bn254 package
+	var g1Aff bn254.G1Affine
+	g1Aff.X.SetOne()
+	g1Aff.Y.SetUint64(2)
```

Comment:
Why is this hard-coded instead of using library function? See https://github.com/consensys/gnark-crypto/blob/v0.16.0/ecc/bn254/bn254.go#L151

Reply:
It’s hardcoded to use the same generators as on the Solidity side. Otherwise, I couldn’t verify the proof.

Reply:
OK, makes sense. But then I think the comment "Retrieve the G1 generator (1,2) from the bn254 package" is misleading.